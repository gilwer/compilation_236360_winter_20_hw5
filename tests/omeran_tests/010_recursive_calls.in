// recursive calls
enum days { SUN, MON, TUES };

void foo(int a) {
  if (a > 0) {
    print("called foo with");
    printi(a);
    foo (a-1);
  }
}

void main() {
  foo(4);
}