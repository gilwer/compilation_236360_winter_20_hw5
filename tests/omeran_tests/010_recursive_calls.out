called foo with
4
called foo with
3
called foo with
2
called foo with
1
